src/util.js

```diff
@@ -254,3 +254,27 @@ export function topTermsFromTranscript(transcript, filterStopWords, limit) {
 
   return filteredTerms;
 }
+
+/**
+ * Finds the actual end time for when a speaker stops talking (collapses multiple of the same speaker in a row into one end time)
+ */
+export function getSpeakerEndTime(segments, segmentIndex) {
+  let initialSegment = segments[segmentIndex];
+  let { endTime } = initialSegment;
+
+  for (let i = segmentIndex; i < segments.length - 1; i++) {
+    const currentSegment = segments[i];
+    const { id: currentSpeakerId } = currentSegment.speakerInfo;
+    endTime = currentSegment.endTime;
+
+    let nextSegment = segments[i + 1];
+    if (
+      !nextSegment.speakerInfo ||
+      nextSegment.speakerInfo.id !== currentSpeakerId
+    ) {
+      return endTime;
+    }
+  }
+
+  return endTime;
+}
```

Comment:
This can be simplified to use const initialSegment and we can just use the speakerInfo there since that has the ID we care about (no need to track the previous / next segment). Just need to track the last seen endTime before the segments[i] has a different ID.